### PR TITLE
feat: add monitoring concept to concepts folder

### PR DESCRIPTION
**Pull request by NkwaTambe**
**See Description here** :  #17

### CODE REVIEW

**Review by Blindspot22 — CHANGES_REQUESTED** [review]
@NkwaTambe Please if I may ask why is the file empty?
And also when you talk of `Design and documented a comprehensive monitoring solution for the online banking application` , in which aspect are you monitoring the project?
1. Checking the performance of the `webank` project, including response times, and error rates. or 
2. Security Monitoring

**Comment by NkwaTambe** [comment]
@Blindspot22 please review new changes..

had issue with the vs code setup of the new pc.

**Review by Blindspot22 — CHANGES_REQUESTED** [review]
@NkwaTambe The term **`Monitoring`** can actually apply to both implementation and documentation, depending on the context of the project.

I believe in the context of the **`webank`** project monitoring involves setting up tools, systems, or processes to actively track and gather data about the performance or health of our system.

This could involve using software's like `Prometheus`, `Grafana`, or `Nagios` as you said in your description above in other to collect and display metrics in real-time. 

So i will like to know is this the documentation for the implementation to be done **`(monitoring-concept-doc)`** or it's the actual **`monitoring concept`** ?

File: docs/concepts/monitoring.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+### b. Grafana
+- **Purpose**: Grafana will be used to visualize the data collected by Prometheus through custom dashboards.
+- **Dashboards**: Create dashboards for different environments (development, staging, production) to provide real-time visibility into system performance.
+- **Key Visualizations**:
+  - CPU and memory usage over time
+  - API request/response rates
+  - Transaction volumes
+  
+
+### c. ELK Stack (Elasticsearch, Logstash, Kibana)
```
Comment (nancymuyeh) [inline docs/concepts/monitoring.md:66]: The ELK stack is primarily focused on logging and we have a ticket for that so this setup should not be here.

**Review by nancymuyeh — CHANGES_REQUESTED** [review]

**Comment by NkwaTambe** [comment]
> @NkwaTambe The term **`Monitoring`** can actually apply to both implementation and documentation, depending on the context of the project.
> 
> I believe in the context of the **`webank`** project monitoring involves setting up tools, systems, or processes to actively track and gather data about the performance or health of our system.
> 
> This could involve using software's like `Prometheus`, `Grafana`, or `Nagios` as you said in your description above in other to collect and display metrics in real-time.
> 
Hello @Blindspot22 , we are not using **Nagios** for now, i did not mension that in my doc like you said

> So i will like to know is this the documentation for the implementation to be done **`(monitoring-concept-doc)`** or it's the actual **`monitoring concept`** ?

it is the documentation for the implementation to be done which is actually the monitoring concept i guess..
please correct me if i am wrong..

File: docs/concepts/monitoring.md (hunk 1 above)
Comment (NkwaTambe) [reply docs/concepts/monitoring.md:66]: Yes yes..

let me correct that now @nancymuyeh

**Review by Blindspot22 — APPROVED** [review]
LGTM  🚀

**Comment by Koufan-De-King** [comment]
> it is the documentation for the implementation to be done which is actually the monitoring concept i guess..
> please correct me if i am wrong..

This is correct, @NkwaTambe

**Review by nancymuyeh — APPROVED** [review]
good job @NkwaTambe

**Pull request merged**